internal/action/command.go

```diff
@@ -1077,52 +1109,17 @@ func (h *BufPane) TermCmd(args []string) {
 		args = []string{sh}
 	}
 
-	term := func(i int, newtab bool) {
-		t := new(shell.Terminal)
-		err := t.Start(args, false, true, nil, nil)
-		if err != nil {
-			InfoBar.Error(err)
-			return
-		}
-
-		id := h.ID()
-		if newtab {
-			h.AddTab()
-			i = 0
-			id = MainTab().Panes[0].ID()
-		} else {
-			MainTab().Panes[i].Close()
-		}
-
-		v := h.GetView()
-		tp, err := NewTermPane(v.X, v.Y, v.Width, v.Height, t, id, MainTab())
-		if err != nil {
-			InfoBar.Error(err)
-			return
-		}
-		MainTab().Panes[i] = tp
-		MainTab().SetActive(i)
-	}
-
 	// If there is only one open file we make a new tab instead of overwriting it
 	newtab := len(MainTab().Panes) == 1 && len(Tabs.List) == 1
-
 	if newtab {
-		term(0, true)
+		h.openTerm(args, true)
 		return
 	}
 
-	for i, p := range ps {
-		if p.ID() == h.ID() {
-			if h.Buf.Modified() && !h.Buf.Shared() {
-				h.closePrompt("Save", func() {
-					term(i, false)
-				})
-			} else {
-				term(i, false)
-			}
-			break
-		}
+	if h.Buf.Modified() && !h.Buf.Shared() {
```

Comment:
What about `else`?
It's a regression (try it).

Reply:
Thanks, got lost in a intermediate step. :man_facepalming: